### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by MMGGYY66**
## OOP school library: add basic UI
- [Create a app.rb file that will serve as your console app entry-point](https://github.com/MMGGYY66/oop-school-lib/commit/74b820925cb4a66a26acbe0d43e025b5102c91bf)
- [Create the App class and initialize method](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/0b19877473d85eb7f4a0aa3d63ca7d2bb4713c2c)
- [create a method to show List all books](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/ca4d6722f391db875b82d9aeff117d90af4c2dad)
- [create a method to show List all people](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/3b34f15a45da1840abdbecb33932a6003444e701)
- [create a method to create a person](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/7ed3b490481ac3fc398425a40aa296202cc04272)
- [create a method to create a book](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/80a71884974eb5ef6c361cc6d1cefb561acf5b96)
- [create a method to create a rental](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/d3087039cefc0f5144482c8e14e3eb0794067952)
- [Create a method to show List all rentals](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/da0ac1f065ce45d2eaa1c9d9e715d43f36e21936)
- [Create methods that should do the following Present the user with a list of options to perform](https://github.com/MMGGYY66/oop-school-lib/pull/4/commits/7d745fb79bbc2d0b45d1ec3e213031d0ac6738d5)

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  puts '3 - Create a person'
+  puts '4 - Create a book'
+  puts '5 - Create a rental'
+  puts '6 - List all rentals for a given person id'
+  puts '7 - Exit'
+end
+
+def main
+  app = App.new
+
```
Comment (Owusu-Desmond) [inline main.rb:17]: - [ ] As per the project requirements, please define the entry point in main.rb. This should be a method called "main" that will be invoked at the end of the file. The main method should perform the following tasks:
  - Display a list of options for the user to choose from.
  - Allow the user to select an option.
  - If necessary, prompt the user for any required parameters.
  - Provide a way for the user to exit the application.
  
  However, please note that when I attempt to start your application using "ruby main.rb", I get this error message below :point_down:  . Please ensure that this issue is resolved and that the app functions correctly according to the requirements.
    
    image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @MMGGYY66 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
📌 No Linters error ✅
📌 clean code :+1:  ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (MMGGYY66) [reply main.rb:17]: it is a propblem in your computer. it is working with me fine and properly. try fix your computer
it is working  in my computer see this screen shot:

main

File: person.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
-require_relative 'name_able'
-require_relative 'trimmer_decorator '
-require_relative 'capital_decorator'
+require './name_able'
+require './trimmer_decorator '
```
Comment (franclobo) [inline person.rb]: - [ ] I had the same error as the last code reviewer 👇🏽 

```ruby
franclobo@DESKTOP-E09US2D:~/microverse/oop-school-lib$ ruby main.rb
<internal:/home/franclobo/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- ./trimmer_decorator  (LoadError)
        from <internal:/home/franclobo/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/franclobo/microverse/oop-school-lib/person.rb:2:in `<top (required)>'
        from /home/franclobo/microverse/oop-school-lib/student.rb:1:in `require_relative'
        from /home/franclobo/microverse/oop-school-lib/student.rb:1:in `<top (required)>'
        from /home/franclobo/microverse/oop-school-lib/app.rb:1:in `require_relative'
        from /home/franclobo/microverse/oop-school-lib/app.rb:1:in `<top (required)>'
        from main.rb:1:in `require_relative'
        from main.rb:1:in `<main>'
```

- [ ] Please 🙏🏽 correct the typo error, there is an extra space at the end of `require './trimmer_decorator '` to be able to use the app without issues. After that, everything is working.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @MMGGYY66, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (MMGGYY66) [reply person.rb]: ok thank you
done

**Review by franclobo — APPROVED** [review]
Hi @MMGGYY66, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️
- Descriptive PR ✔️
- Correct Git flow ✔️
- Good UI ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MMGGYY66** [comment]
Thank you

**Pull request merged**